odoo/tools/misc.py

```diff
@@ -135,124 +135,66 @@ def exec_pg_command_pipe(name, *args):
 #----------------------------------------------------------
 # File paths
 #----------------------------------------------------------
-#file_path_root = os.getcwd()
-#file_path_addons = os.path.join(file_path_root, 'addons')
 
-def file_open(name, mode="r", subdir='addons', pathinfo=False):
-    """Open a file from the OpenERP root, using a subdir folder.
+def file_path(file_path, filter_ext=None):
+    """Verify that a file exists under a known `addons_path` directory and return its full path.
 
     Example::
 
-    >>> file_open('hr/report/timesheer.xsl')
-    >>> file_open('addons/hr/report/timesheet.xsl')
+    >>> file_path('hr')
+    >>> file_path('hr/static/description/icon.png')
 
-    @param name name of the file
-    @param mode file open mode
-    @param subdir subdirectory
-    @param pathinfo if True returns tuple (fileobject, filepath)
-
-    @return fileobject if pathinfo is False else (fileobject, filepath)
+    :param str file_path: absolute file path, or relative path within any `addons_path` directory
+    :param list(str) filter_ext: optional list of supported extensions (without leading dot)
```

Comment:
Although sphinx supports both parenthetical and bracketed forms, the bracketed form is probably better for compatility with `typing`.